drivers/include/periph/gpio_ll.h

```diff
@@ -0,0 +1,498 @@
+ *
+ * It is fully valid that an implementation extends this structure with
+ * additional implementation specific fields. For example, it could be useful
+ * to also include fields to configure routing of a GPIO pin to other
+ * peripherals (e.g. for us as an TXD pin of an UART). These implementation
+ * specific fields **MUST** however have reasonable defaults when initialized
+ * with zero (e.g. pin is not routed to another peripheral but to be used as
+ * regular GPIO). For obvious reasons, portable code cannot rely on the
+ * presence and semantic of any implementation specific fields.  Additionally,
+ * out-of-tree users should not use these fields, as the implementation
+ * specific fields cannot be considered a stable API.
+ */
+typedef struct {
+    gpio_state_t state;         /**< State of the pin */
+    gpio_pull_t pull;           /**< Pull resistor configuration */
+    /**
+     * @brief   Configure the slew rate of outputs
+     *
+     * This value is ignored *unless* @ref gpio_conf_t::state is configured
+     * to @ref GPIO_OUTPUT_PUSH_PULL or @ref GPIO_OUTPUT_OPEN_DRAIN. A call to
+     * @ref gpio_ll_init will update this field to the configuration that was
+     * actually used. A caller relying on a specific pull resistor must hence check
+     * if the configuration was applied as requested.
+     */
+    gpio_slew_t slew_rate;
+    /**
+     * @brief   Whether to enable the input Schmitt trigger
+     *
+     * This value is ignored *unless* @ref gpio_conf_t::state is configured
+     * to @ref GPIO_INPUT. A call to @ref gpio_ll_init will update this field
+     * to the configuration that was actually used. A caller relying on a
+     * specific Schmitt trigger configuration must hence check if the
+     * configuration was applied as requested.
+     */
+    bool schmitt_trigger;
+    /**
+     * @brief   Initial value of the output
+     *
+     * Ignored if @ref gpio_conf_t::state is set to @ref GPIO_INPUT or
+     * @ref GPIO_DISCONNECT. If the pin was previously in a high impedance
+     * state, it is guaranteed to directly transition to the given initial
+     * value.
+     */
+    bool initial_value;
+    /**
+     * @brief   Strength of the pull up/down resistor
+     *
+     * This value is ignored when @ref gpio_conf_t::pull is configured to
+     * @ref GPIO_FLOATING. A call to @ref gpio_ll_init will select the pull
+     * resistor closet to the specified value and update this field to the
+     * actually used value.
+     */
+    gpio_pull_strength_t pull_strength;
+    /**
+     * @brief   Drive strength of the GPIO
+     *
+     * This value is ignored when @ref gpio_conf_t::state is configured to
+     * @ref GPIO_INPUT or @ref GPIO_DISCONNECT. A call to @ref gpio_ll_init
+     * will select the drive strength closet to the specified value and update
+     * this field to the actually used value.
+     */
+    gpio_drive_strength_t drive_strength;
+} gpio_conf_t;
```

Comment:
I'm sure we can provide some convenience defaults for this

```C
extern const gpio_conf_t *GPIO_LL_IN;
extern const gpio_conf_t *GPIO_LL_IN_PU;
extern const gpio_conf_t *GPIO_LL_IN_PD;
extern const gpio_conf_t *GPIO_LL_OUT_LOW;
extern const gpio_conf_t *GPIO_LL_OUT_HIGH;
…
```

Reply:
For me, it seems so.